foundations/javascript_basics/fundamentals-5.md

```diff
@@ -6,30 +6,153 @@ Congratulations on making it to one of the last lessons in Foundations! By this
+
+#### The reduce method
+
+Finally, let's say that we wanted to multiply all of the numbers in our `arr` together like this: `1 * 2 * 3 * 4 * 5`.
+First, we'd have to declare a variable `total` and initialize it to 1. Then, we'd iterate through the array with a `for` loop and multiply the `total` by the current number.
+
+But we don't actually need to do all of that, we have our `reduce` method that will do the job for us. Just like `.map()` and `.reduce()` it expects a callback function.
+However, there are two key differences with this array method:
+- The callback function takes two arguments instead of one. The first argument is the `accumulator`, which is the current value of the result *at that point in the loop*. The first time through, this value will either be set to the `initialValue` (described in the next bullet point), or the first element in the array if no `initialValue` is provided. The second argument for the callback is the `current` value, which is the item currently being iterated on.
+- It also takes in an `initialValue` as a second argument (after the callback), which helps when we don't want our initial value to be the first element in the array. For instance, if we wanted to sum all numbers in an array, we could call reduce without an `initialValue`, but if we wanted to sum all numbers in an array and add 10, we could use 10 as our `initialValue`.
+
+```js
+arr.reduce(((total, currentItem) => total * currentItem), 1); // Outputs [1, 2, 6, 24, 120];
+```
+
+In the above function, we: 
+- Pass in a callback function, which is `(total, currentItem) => total * currentItem`.
+- Initialize total to `1` in the second argument.
+
+So what `.reduce()` will do, is it will once again go through every element in `arr` and apply the `callback` function to it. It then changes `total`, without actually changing the array itself. After it's done, it returns `total`.
+
+#### Summary 
+
+You've learnt about the three powerful array methods which are `map`, `filter` and `reduce`. They allow us to write shorter code that is more readable and less prone to bugs. 
+
+For a quick recap of these array methods, consider this picture which should visually explain them in terms of sandwiches:
+
+![example of filter, map and reduce methods being used to visually represent making a sandwich](https://static.observableusercontent.com/thumbnail/bea194824f0d5842addcb7910bb488795c6f80f143ab5332b28a317ebcecd603.jpg)
+
+Let's do some quick practice before your assignment! Rewrite the `sumOfTripledEvens(array)` function using these three methods.
+
+Once you are finished and you've tested that your function works correctly, check out the solution below.
+<details><summary>Solution</summary>
+
+```javascript
+function sumOfTripledEvens(array) {
+  return array
+    .filter(num => num % 2 === 0)
+    .map(num => num * 3)
+    .reduce((acc, curr) => acc + curr);
+}
+```
+</details>
```

Comment:
I don't think the backticks inside the details block is going to work, at least according to the [lesson preview tool](https://www.theodinproject.com/lessons/preview). My suggestion of using the code elements might though. @thatblindgeye do you have any suggestions regarding this?

Reply:
You need to give the `details` element the attribute `markdown="block"`
```md
<details markdown="block">
</details>
```
Then it will render the markdown properly.

Reply:
I've added 

> You need to give the `details` element the attribute `markdown="block"`
> 
> ```
> <details markdown="block">
> </details>
> ```
> 
> Then it will render the markdown properly.

Added it, awesome. It works correctly now.